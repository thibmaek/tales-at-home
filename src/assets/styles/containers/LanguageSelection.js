import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: `center`,
  },
  title: {
    textAlign: `center`,
    color: `white`,
    fontSize: 24,
    backgroundColor: `transparent`,
  },
});
