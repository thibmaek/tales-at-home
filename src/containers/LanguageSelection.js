import React, { Component, PropTypes } from 'react';
import { View, Picker, Text } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import s from '../styles/LanguageSelection';

export default class App extends Component {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <LinearGradient colors={[`#3fa7e6`, `#4cccf1`]}  style={s.container}>
        <View style={s.view}>
          <Text style={s.title}>Kies een taal</Text>
          <Picker
            selectedValue={this.props.initialLang}
            onValueChange={lang => this.setState({ lang })}
          >
            <Picker.Item label='English' value='en' />
            <Picker.Item label='Nederlands' value='nl' />
          </Picker>
        </View>
      </LinearGradient>
    );
  }
}

App.propTypes = {
  initialLang: PropTypes.string.isRequired,
};
