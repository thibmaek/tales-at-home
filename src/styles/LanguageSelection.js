import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: `center`,
  },
  title: {
    textAlign: `center`,
    fontSize: 24,
  },
});



export default styles;
