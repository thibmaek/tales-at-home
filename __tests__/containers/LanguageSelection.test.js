import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import LanguageSelection from '../../src/containers/LanguageSelection';

jest.mock(`react-native-linear-gradient`, () => `LinearGradient`);

describe(`Container: <LanguageSelection />`, () => {
  it(`should render the container`, () => {
    const tree = renderer.create(<LanguageSelection />).toJSON();
    expect(tree).toMatchSnapshot(`LanguageSelection`);
  });

  it(`should not accept {...initialLang<Any>} as a prop`, () => {
    const tree = renderer.create(<LanguageSelection initialLang={4} />).toJSON();
    expect(tree).toThrowError();
  });
});
