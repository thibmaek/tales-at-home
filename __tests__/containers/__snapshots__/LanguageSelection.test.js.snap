exports[`LanguageSelection 1`] = `
<LinearGradient
  colors={
    Array [
      "#3fa7e6",
      "#4cccf1",
    ]
  }
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 24,
          "textAlign": "center",
        }
      }>
      Kies een taal
    </Text>
    <View
      style={undefined}>
      <RCTPicker
        items={
          Array [
            Object {
              "label": "English",
              "textColor": undefined,
              "value": "en",
            },
            Object {
              "label": "Nederlands",
              "textColor": undefined,
              "value": "nl",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        } />
    </View>
  </View>
</LinearGradient>
`;
